src/common/history.c

```diff
@@ -956,21 +968,31 @@ void dt_history_compress_on_image(int32_t imgid)
     sqlite3_finalize(stmt);
   }
 
-  // Update XMP files
-  dt_image_synch_xmp(imgid);
-}
+  // load new history and write it back to ensure that all history are properly numbered without a gap
+  dt_dev_reload_history_items(darktable.develop);
 
-void dt_history_compress_on_image_and_reload(int32_t imgid)
-{
-  dt_history_compress_on_image(imgid);
+  // then we can get the item to select in the new clean-up history retrieve the position of the module
+  // corresponding to the history end.
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "SELECT IFNULL(MAX(num)+1, 0) FROM main.history "
+                                                             "WHERE imgid=?1", -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
 
-  /* if current image in develop reload history */
-  if(dt_dev_is_current_image(darktable.develop, imgid))
-  {
-    dt_dev_reload_history_items(darktable.develop);
-    dt_dev_write_history(darktable.develop);
-    dt_dev_modulegroups_set(darktable.develop, dt_dev_modulegroups_get(darktable.develop));
```

Comment:
Difficult to understand from the diff, but it seems that the call dt_dev_modulegroups_set() is completely removed. Is that really intended? Was this call superfluous as already taken care in another routine? Just want to be sure this is not an oversight.

Reply:
Thanks for observing this, but do we need this for every picture after compression or only if it is in darktable mode?

Reply:
Activated always. No problem.